tests/std/tests/P1614R2_spaceship/test.cpp

```diff
@@ -54,29 +58,105 @@ struct SynthOrdered {
     }
 };
 
-template <class T>
-inline constexpr bool is_pair = false;
-template <class A, class B>
-inline constexpr bool is_pair<std::pair<A, B>> = true; // TRANSITION, std::pair spaceship not yet implemented
+struct OrderedChar {
+    OrderedChar() = default;
+    OrderedChar(const char c) : c(c){};
 
-template <class Container>
-void ordered_containers_test(const Container& smaller, const Container& smaller_equal, const Container& larger) {
+    OrderedChar& operator=(const char& other) {
+        this->c = other;
+        return *this;
+    }
+
+    operator char() {
+        return c;
+    }
+
+    char c;
+};
+
+auto operator<=>(const OrderedChar& left, const OrderedChar& right) {
+    return left.c <=> right.c;
+}
+
+auto operator==(const OrderedChar& left, const OrderedChar& right) {
+    return left.c == right.c;
+}
+
+struct WeaklyOrderedChar : public OrderedChar {};
+struct WeaklyOrderdByOmissionChar : public OrderedChar {};
+struct PartiallyOrderedChar : public OrderedChar {};
+
+namespace std {
+    template <>
+    struct char_traits<OrderedChar> : public std::char_traits<char> {
+        using char_type = OrderedChar;
+
+        static int compare(const char_type* first1, const char_type* first2, size_t count) {
+            for (; 0 < count; --count, ++first1, ++first2) {
+                if (*first1 != *first2) {
+                    return *first1 < *first2 ? -1 : +1;
+                }
+            }
+
+            return 0;
+        }
+
+        static bool eq(const char_type l, const char_type r) {
+            return l.c == r.c;
+        }
+    };
+
+    template <>
+    struct char_traits<WeaklyOrderedChar> : public std::char_traits<OrderedChar> {
+        using char_type           = WeaklyOrderedChar;
+        using comparison_category = std::weak_ordering;
+    };
+
+    template <>
+    struct char_traits<WeaklyOrderdByOmissionChar> : public std::char_traits<OrderedChar> {
+        using char_type = WeaklyOrderdByOmissionChar;
+    };
+
+    template <>
+    struct char_traits<PartiallyOrderedChar> : public std::char_traits<OrderedChar> {
+        using char_type           = PartiallyOrderedChar;
+        using comparison_category = std::partial_ordering;
+    };
+} // namespace std
+
+template <class SmallType, class EqualType, class LargeType, class ReturnType>
+void spaceship_test(const SmallType& smaller, const EqualType& smaller_equal, const LargeType& larger) {
+    assert(smaller == smaller_equal);
+    assert(smaller != larger);
     assert(smaller < larger);
-    assert(smaller <= larger);
     assert(larger > smaller);
+    assert(smaller <= larger);
     assert(larger >= smaller);
-    assert(smaller == smaller_equal);
-    assert(smaller != larger);
     assert((smaller <=> larger) < 0);
     assert((larger <=> smaller) > 0);
     assert((smaller <=> smaller_equal) == 0);
 
+    static_assert(std::is_same_v<decltype(smaller <=> larger), ReturnType>);
+}
+
+template <class TestType, class ReturnType>

```

Comment:
Do we really need both  overloads of `spaceship_test`, or could we put `ReturnType` first - to allow it to be explicitly specified` - and get away with just the first overload by passing arguments of the intended types?